### DISCUSSION TITLE
Automated snapshots

### DISCUSSION THREAD

**Discussion by timwhite1** [stratis-storage/project#542] [Q&A]
Has anyone tested using Systemd timers, cron jobs, or other method to take automated Stratis snapshots and cull at a given expiration interval?

I would like to setup the following:
Daily - retain 10
Weekly - retain 4
Monthly - retain 6

**Comment by jbaublitz** [comment]
We've seen a report of someone doing this before, but I'm uncertain of how they did the expiration. If you're willing to write a Python script or something similar to access our D-Bus API, there should be enough information if you both name the snapshots appropriately and access the creation time as a D-Bus property. Then you should be able to sort the creation time and delete everything but the last `n` snapshots.

**Reply by timwhite1** [reply]
That may be a little intense for me but thank you for the info! I appreciate your response.
